File: src/cross_read/models/files.py
from __future__ import annotations

from datetime import datetime
from enum import StrEnum

from pydantic import BaseModel


class FileKind(StrEnum):
    DIRECTORY = "directory"
    MARKDOWN = "markdown"
    PDF = "pdf"
    DOCX = "docx"
    IMAGE = "image"
    TEXT = "text"
    VIDEO = "video"
    UNSUPPORTED = "unsupported"


class ShareSummary(BaseModel):
    id: str
    name: str


class ShareListResponse(BaseModel):
    items: list[ShareSummary]


class FileEntry(BaseModel):
    name: str
    path: str
    kind: FileKind
    is_directory: bool
    size: int | None
    modified_at: datetime


class DirectoryResponse(BaseModel):
    share: ShareSummary
    path: str
    items: list[FileEntry]


class StatusResponse(BaseModel):
    name: str
    version: str
    status: str

File: frontend/src/types/files.ts
export type FileKind =
  | 'directory'
  | 'markdown'
  | 'pdf'
  | 'docx'
  | 'spreadsheet'
  | 'presentation'
  | 'image'
  | 'text'
  | 'audio'
  | 'video'
  | 'unsupported'

export interface ShareSummary {
  id: string
  name: string
}

export interface ShareListResponse {
  items: ShareSummary[]
}

export interface FileEntry {
  name: string
  path: string
  kind: FileKind
  is_directory: boolean
  size: number | null
  modified_at: string
}

export interface DirectoryResponse {
  share: ShareSummary
  path: string
  items: FileEntry[]
}

export interface SearchResponse {
  share: ShareSummary
  path: string
  query: string
  items: FileEntry[]
  truncated: boolean
}
